pepper-apis/src/main/java/org/broadinstitute/ddp/customexport/export/CustomExportCoordinator.java

```diff
@@ -0,0 +1,346 @@
+import com.typesafe.config.Config;
+import org.broadinstitute.ddp.client.ApiResult;
+import org.broadinstitute.ddp.client.SendGridClient;
+import org.broadinstitute.ddp.customexport.CustomActivityExtract;
+import org.broadinstitute.ddp.customexport.constants.CustomExportConfigFile;
+import org.broadinstitute.ddp.customexport.db.dao.CustomExportDao;
+import org.broadinstitute.ddp.db.dto.StudyDto;
+import org.broadinstitute.ddp.exception.DDPException;
+import org.broadinstitute.ddp.model.study.Participant;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CustomExportCoordinator {
+
+    private static final Logger LOG = LoggerFactory.getLogger(CustomExportCoordinator.class);
+
+    private final CustomExporter exporter;
+    private Bucket csvBucket;
+    private final String customActivity;
+    private final String customExportStatus;
+    private final Config exportCfg;
+    private String fullFileName;
+
+    public CustomExportCoordinator(Config exportCfg) {
+        this.exportCfg = exportCfg;
+        this.customActivity = exportCfg.getString(CustomExportConfigFile.ACTIVITY);
+        this.customExportStatus = exportCfg.getString(CustomExportConfigFile.STATUS);
+        this.exporter = new CustomExporter(exportCfg);
+    }
+
+    public CustomExportCoordinator includeCsv(Bucket csvBucket) {
+        this.csvBucket = csvBucket;
+        return this;
+    }
+
+    public boolean exportCustom(StudyDto customDto) {
+        SendGridClient client =
+                new SendGridClient(exportCfg.getConfig(CustomExportConfigFile.EMAIL)
+                        .getString(CustomExportConfigFile.EMAIL_SENDGRID_TOKEN));
+
+        //Find out whether there is anything to export
+        boolean needExport = withAPIsTxn(handle -> {
+            CustomExportDao customExportDao = handle.attach(CustomExportDao.class);
+            Instant lastCompleted = customExportDao.getLastCompleted(customDto.getId());
+            return customExportDao.needCustomExport(customDto.getId(), customExportStatus, lastCompleted.toEpochMilli(), customActivity);
+        });
+
+        if (!needExport) {
+            LOG.info("Skipping custom export: nothing to export");
+            ApiResult res = client.sendMail(createNotificationEmail(false, true));
+            if (res.hasThrown()) {
+                LOG.error("Sending notification email for study {} with skipped custom export failed", customDto.getGuid(),
+                        res.getThrown());
+            } else if (res.hasError()) {
+                LOG.error("Sending notification email for study {} with skipped custom export failed", customDto.getGuid(),
+                        res.getError());
+            } else if (res.hasFailure()) {
+                LOG.error("Sending notification email for study {} with skipped custom export failed with status code {}",
+                        customDto.getGuid(), res.getStatusCode());
+            }
```

Comment:
I don't think we need a `hasFailure()` branch. We should log the status code in the `hasError()` branch.